libs/grok/src/main/java/org/elasticsearch/grok/Grok.java

```diff
@@ -219,12 +217,10 @@ public boolean match(String text) {
         Matcher matcher = compiledExpression.matcher(text.getBytes(StandardCharsets.UTF_8));
         int result;
         try {
-            threadWatchdog.register();
-            result = matcher.searchInterruptible(0, text.length(), Option.DEFAULT);
-        } catch (InterruptedException e) {
-            result = Matcher.INTERRUPTED;
+            matcherWatchdog.register(matcher);
+            result = matcher.search(0, text.length(), Option.DEFAULT);
         } finally {
-            threadWatchdog.unregister();
+            matcherWatchdog.unregister(matcher);
         }
         return (result != -1);
```

Comment:
> But it might be best to make the return value more intuitive before someone else uses this method in production code.

Agreed. I will change the jdocs in this PR and in a followup will do the change that you're suggesting here.